File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        if not nums:
            return False
        n = len(nums)
        if n <= 1:
            return False

        hashmap = {}
        for i in range(n):
            if nums[i] in hashmap:
                return True
            hashmap[nums[i]] = i
        return False

    

        

File: Data Structures & Algorithms/missing-number/submission-1.java
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int xor = 0;

        for (int i = 0; i <= n; i++) {
            xor ^= i;
        }

        for (int num : nums) {
            xor ^= num;
        }
        return xor;
    }
}
        // n = len(nums)
        // xor = 0
        
        // # XOR all numbers from 0 to n
        // for i in range(n + 1):
        //     xor ^= i
        
        // # XOR with all array elements
        // for num in nums:
        //     xor ^= num
        
        // return xor